lesson18/index.js

```diff
@@ -166,9 +167,34 @@ const addEventListenerForPagination = (dataLength) => {
         switchPagination(clickedItemIndex);
         incrementCurrentIndex(clickedItemIndex);
         toggleButtonDisabled(dataLength);
+        initAutoMoveSlide(dataLength);
     })
 }
 
+//タイマー停止用のID
+let intervalId;
+
+function autoMoveSlide(dataLength) {
+
+    intervalId = setInterval(() => {
+        let currentIndex = getCurrentIndex();
+        currentIndex++;
+    
+        if (currentIndex === dataLength) {
+            currentIndex = 0;
+        }
+    
+        switchImg(currentIndex);
+        switchPagination(currentIndex);
+        incrementCurrentIndex(currentIndex);
+        toggleButtonDisabled(dataLength);
+    }, 3000);
+}
```

Comment:
IMO
Rather than defining a let globally, why not think of it in the form of an assignment to an object?
```suggestion
//タイマー停止用のID
const intervalCount = { count: 0 };
function autoMoveSlide(dataLength) {
  intervalCount.count = setInterval(() => {
    let currentIndex = getCurrentIndex();
    currentIndex++;
    if (currentIndex === dataLength) {
      currentIndex = 0;
    }
    switchImg(currentIndex);
    switchPagination(currentIndex);
    incrementCurrentIndex(currentIndex);
    toggleButtonDisabled(dataLength);
  }, 3000);
}
```

Reply:
Thanks for the review.
I hadn't thought of that at all ...!
Instead of assigning to a let-defined ID, you assign to a const-defined object.
I learned a lot!
[3971450](https://github.com/haru-programming/JavaScript-lessons/pull/23/commits/3971450b39526bf7085d85a2f86c5cb793cfc6c5)